homeassistant/components/zwave_js/update.py

```diff
@@ -187,20 +188,26 @@ async def _async_update(self, _: HomeAssistant | datetime | None = None) -> None
                 err,
             )
         else:
-            if available_firmware_updates:
-                self._latest_version_firmware = latest_firmware = max(
-                    available_firmware_updates,
-                    key=lambda x: AwesomeVersion(x.version),
+            # If we have an available firmware update that is a higher version than
+            # what's on the node, we should advertise it, otherwise the installed
+            # version is the latest.
+            if (
+                available_firmware_updates
+                and (
+                    latest_firmware := max(
+                        available_firmware_updates,
+                        key=lambda x: AwesomeVersion(x.version),
+                    )
                 )
-
-                # If we have an available firmware update that is a higher version than
-                # what's on the node, we should advertise it, otherwise there is
-                # nothing to do.
-                new_version = latest_firmware.version
-                current_version = self.node.firmware_version
-                if AwesomeVersion(new_version) > AwesomeVersion(current_version):
-                    self._attr_latest_version = new_version
-                    self.async_write_ha_state()
+                and AwesomeVersion(latest_firmware.version)
```

Comment:
```suggestion
                and latest_firmware.version
```

This is already `AwesomeVersion` object from line 199

Reply:
I accepted this but I don't actually think that's true. We create an awesome version in the lambda but the return from the max is the original object

Reply:
let's see what the tests say

Reply:
yep tests failed, I removed this commit

Reply:
tests pass but I think my logic is correct?

Reply:
Yeah, sorry my bad..
Your logic is correct on both counts 👍